app/services/importers/okc_blotter/pdf.rb

```diff
@@ -0,0 +1,139 @@
+require 'zip'
+
+module Importers
+  module OkcBlotter
+    class Pdf < ApplicationService
+      def initialize(date)
+        @date = date
+      end
+
+      def perform
+        download_pdf_from_website unless pdf
+        save
+      end
+
+      def download_pdf_from_website
+        input = HTTParty.get("#{url}/pdfs?date=#{@date}",
+                             headers: {
+                               Authorization: auth_token
+                             }).body
+        Zip::InputStream.open(StringIO.new(input)) do |io|
+          while entry = io.get_next_entry #note: there is only actually one pdf here
+            pdf = io.read
+            filename = entry.name
+            Bucket.new.put_object("#{s3_path}/#{filename}", pdf)
+          end
+        end
+      end
+
+      def save
+        bookings_hash = json
+        ::OkcBlotter::Pdf.create!(
+          parsed_on: DateTime.now,
+          date: @date,
+          bookings: bookings_hash.map do |booking_hash|
+            build_bookings(booking_hash)
+          end
+        )
+      end
+
+      def json
```

Comment:
A little confused on what this does after reading the hack.

Reply:
we're POSTing a PDF to the endpoint. We're just storing the PDF in memory but HTTParty only supports it if we're saving the PDF first to the local file system.  There's no need for us to do that since we're already saving it to s3, but HTTPparty is expecting it to have a path parameter indicating that we're passing in a file. This is a hack to get around that.

Reply:
Ok, I think I understand, would love to half a walkthrough of this though.